workshops/filters.py

```diff
@@ -261,6 +261,25 @@ def filter_trainees_by_training_request_presence(queryset, flag):
         return queryset.filter(trainingrequest__isnull=True)
 
 
+def filter_trainees_by_instructor_status(queryset, choice):
+    if choice == '':
+        return queryset
+    elif choice == 'swc-and-dc':
+        return queryset.filter(is_swc_instructor=True, is_dc_instructor=True)
+    elif choice == 'swc-or-dc':
+        return queryset.filter(Q(is_swc_instructor=True) |
+                               Q(is_dc_instructor=True))
+    elif choice == 'swc':
+        return queryset.filter(is_swc_instructor=True)
+    elif choice == 'dc':
+        return queryset.filter(is_dc_instructor=True)
+    elif choice == 'eligible':
+        return queryset.filter(Q(swc_eligible=True) | Q(dc_eligible=True),
+                               is_swc_instructor=False, is_dc_instructor=False)
```

Comment:
This would end up filtering out people with an SWC badge but still eligible for obtaining a DC badge.

Is that the intended functionality?